src/config-array.js

```diff
@@ -509,6 +514,8 @@ export class ConfigArray extends Array {
 
 		for (const config of this) {
 
+			assertValidFilesAndIgnores(config);
+
```

Comment:
It looks like this is the only place that's validating files/ignores? I understand from the way the code is currently written that this will effectively be called before `getConfig()`, but that's relying on an implementation detail that could change in the future, nevermind being a bit confusing for people trying to learn (or re-learn) the codebase.

What do you think about changing `assertValidFilesAndIgnores` to cache the validations and then also call it elsewhere, maybe in `getConfig()`?

Reply:
Updated in 5108e26eedf7790197dd168181e5c50ce96be1da, thanks! Definitely a cleaner approach to call the validation function in `getConfig()` beforehand. I've added a flag to the `dataCache` data so we can keep track of config arrays that have already passed the `files`-and-`ignores` validation, and not validate them again.